### PR TITLE
L2 Cache Maintenance in common code

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

File: core/arch/arm/mm/core_mmu.c [hunk 1]
```diff
@@ -431,14 +431,51 @@ unsigned int cache_maintenance_l1(int op, void *va, size_t len)
 	return TEE_SUCCESS;
 }
 
-__weak unsigned int cache_maintenance_l2(int op __unused,
-			paddr_t pa __unused, size_t len __unused)
+#ifdef CFG_PL310
+#include <kernel/tz_ssvce_pl310.h>
```
Comment (jbech-linaro) [inline core/arch/arm/mm/core_mmu.c]: In another patch you agreed with me to have the include at the top of the file. I'd guess that would make sense here also? (However in this case I think we should still put the under the ifdef).

File: core/arch/arm/mm/core_mmu.c [hunk 2]
```diff
@@ -431,14 +431,51 @@ unsigned int cache_maintenance_l1(int op, void *va, size_t len)
...
+	tee_l2cc_mutex_unlock();
+	thread_set_exceptions(exceptions);
+	return ret;
+}
+#else
+unsigned int cache_maintenance_l2(int op __unused,
+				   paddr_t pa __unused, size_t len __unused)
+{
+	/* Nothing to do about L2 Cache Maintenance when no PL310 */
+	return TEE_SUCCESS;
```
Comment (jforissier) [inline core/arch/arm/mm/core_mmu.c]: Better define this this as static inline in `core/arch/arm/include/mm/core_mmu.h` when CFG_PL310 is not set

File: core/arch/arm/mm/core_mmu.c [hunk 3]
```diff
@@ -431,14 +431,51 @@ unsigned int cache_maintenance_l1(int op, void *va, size_t len)
 	return TEE_SUCCESS;
 }
 
-__weak unsigned int cache_maintenance_l2(int op __unused,
-			paddr_t pa __unused, size_t len __unused)
+#ifdef CFG_PL310
+#include <kernel/tz_ssvce_pl310.h>
+#include <kernel/tee_l2cc_mutex.h>
+#include <kernel/thread.h>
```
Comment (jforissier) [inline core/arch/arm/mm/core_mmu.c]: Move these at the beginning of the file?

**Comment by jbech-linaro** [comment]
When comments has been addressed
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by pascal-brand-st-dev** [comment]
Review comments have been addressed

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Pull request merged**